src/schema/browser-apis.js

```diff
@@ -59,18 +160,47 @@ export function isDeprecatedApi(namespace, property) {
   );
 }
 
-export function hasBrowserApi(namespace, property) {
-  const schema = schemas[namespace];
+export function hasBrowserApi(
+  namespace,
+  property,
+  addonMetadata,
+  apiSchemas = schemas
+) {
+  const schema = apiSchemas[namespace];
+
   // We "have" the API if it's deprecated or temporary so we don't double warn.
   if (
     isTemporaryApi(namespace, property) ||
-    isDeprecatedApi(namespace, property)
+    isDeprecatedApi(namespace, property, addonMetadata, apiSchemas)
   ) {
     return true;
   }
 
   const schemaItem =
     getObjectProperty(schema, property) || getArrayProperty(schema, property);
 
+  const manifestVersion = getManifestVersion(addonMetadata);
+  const min_manifest_version = getMinManifestVersion(
+    namespace,
+    property,
+    addonMetadata,
+    apiSchemas
+  );
+  const max_manifest_version = getMinManifestVersion(
+    namespace,
+    property,
+    addonMetadata,
+    apiSchemas
+  );
+  if (
+    manifestVersion < min_manifest_version ||
+    manifestVersion > max_manifest_version
+  ) {
+    // Consider the API unsupported if its schema has a min_manifest_version
+    // greater then the extension manifest version or a max_manifest_version
+    // lower than the extension manifest version.
+    return false;
+  }
```

Comment:
This whole new addition looks similar to the one above (in `isDeprecatedApi()`).

Reply:
@willdurand,
Yeah it is similar, but `isDeprecatedApi` does also have other conditions at the end of it.

In the meantime I have moved the shared check into its own helper function named `isInSupportedManifestVersionRange`, the two checks were similar but with small differences that were not an issue but also not needed, better to have one helper function that we make us sure we do use the exact same code when the check we want is exactly the same one.

As an additional side note, in one of the intermediate versions of this PR I was getting confused by how the `isDeprecatedApi` and `hasBrowserApi` has interacting with each other and I had to fix that, and I also added some inline comments hoping to make that a bit more clear.
To summatize the thing that was confusing me was:

- `hasBrowserApi` has an initial check that basically says: if an API is temporary or is deprecated, then we "have" the API and `hasBrowserApi` should return true to avoid to warn about the same API twice

Tying to clarify that for my "future ourselves" I have added the following two inline comments:

```
export function isDeprecatedApi(...) {
  // If the API has been removed in a certain manifest version, or only
  // available starting from a manifest version, then it is unsupported
  // and not deprecated.
  ...
}

export function hasBrowserApi(...) {
  // We "have" the API if it's deprecated or temporary so we don't double warn.
  ....
  // We don't have the API if the extension manifest_version is outside of the 
  // manifest_version range where the API is actually supported for.
  ....}
  // Or the schema entry for the API has an unsupported property set to true.
  ...
}
```

I'll push an update asap, but I wanted to mention this to you, to double-check explicitly with you if for another engineer perspective (you got extracted as the "lucky one" :-P sorry ;-)) they are of any help.

Reply:
Given that I didn't know about these two functions, it's super helpful!